src/openrct2/object/PeepAnimationsObject.cpp

```diff
@@ -0,0 +1,226 @@
+
+#include "PeepAnimationsObject.h"
+
+#include "../Context.h"
+#include "../PlatformEnvironment.h"
+#include "../core/EnumMap.hpp"
+#include "../core/Guard.hpp"
+#include "../core/Json.hpp"
+#include "../peep/PeepAnimations.h"
+#include "../rct12/RCT12.h"
+#include "ObjectRepository.h"
+
+using namespace OpenRCT2;
+
+static const EnumMap<AnimationPeepType> animationPeepTypeMap({
+    { "guest", AnimationPeepType::Guest },
+    { "handyman", AnimationPeepType::Handyman },
+    { "mechanic", AnimationPeepType::Mechanic },
+    { "security", AnimationPeepType::Security },
+    { "entertainer", AnimationPeepType::Entertainer },
+});
+
+void PeepAnimationsObject::Load()
+{
+    auto numImages = GetImageTable().GetCount();
+    if (numImages == 0)
+        return;
+
+    _imageOffsetId = LoadImages();
+
+    // Set loaded image offsets for all animations
+    for (auto groupKey = 0u; groupKey < _animationGroups.size(); groupKey++)
+    {
+        auto& group = _animationGroups[groupKey];
+        auto& requiredAnimationMap = getAnimationsByPeepType(_peepType);
+        for (auto& [typeStr, typeEnum] : requiredAnimationMap)
+        {
+            group[typeEnum].base_image = _imageOffsetId + group[typeEnum].imageTableOffset;
+            group[typeEnum].bounds = inferMaxAnimationDimensions(group[typeEnum]);
+
+            // Balloons, hats and umbrellas are painted separately, so the inference
+            // algorithm doesn't account for those. We manually compensate for these here.
+            // Between 8-12 pixels are needed, depending on rotation, so we're generalising.
+            auto pag = PeepAnimationGroup(groupKey);
+            if (pag == PeepAnimationGroup::Balloon || pag == PeepAnimationGroup::Hat || pag == PeepAnimationGroup::Umbrella)
+            {
+                group[typeEnum].bounds.sprite_height_negative += 12;
+            }
+        }
+    }
+}
+
+void PeepAnimationsObject::Unload()
+{
+    UnloadImages();
+}
+
+void PeepAnimationsObject::ReadJson(IReadObjectContext* context, json_t& root)
+{
+    Guard::Assert(root.is_object(), "PeepAnimationsObject::ReadJson expects parameter root to be an object");
+    PopulateTablesFromJson(context, root);
+
+    Guard::Assert(root["properties"].is_object(), "PeepAnimationsObject::ReadJson expects properties to be an object");
+    ReadProperties(root["properties"]);
+
+    auto& requiredAnimationMap = getAnimationsByPeepType(_peepType);
+    _animationGroups.clear();
+
+    Guard::Assert(root["animationGroups"].is_array(), "PeepAnimationsObject::ReadJson expects animationGroups to be an array");
+    for (auto& groupJson : root["animationGroups"])
+    {
+        Guard::Assert(groupJson["animations"].is_object(), "PeepAnimationsObject::ReadJson expects animations to be an array");
+
+        PeepAnimations group = ReadAnimations(requiredAnimationMap, groupJson["animations"]);
+
+        // Is this animation group replacing a legacy group?
```

Comment:
Again, already clear from the code itself.

Comments should **add** something, rather than stating the obvious.

Reply:
Haha, I think we just have a different code style here. Personally, I like to have narrative comments in my code, as I think it makes it easier to read for newcomers. As such, I think they do add something.

I checked with some people on Discord, and it seems it's clearly not common for this codebase. I'll remove the 'obvious' comments you've pointed out.